lib/orbit_web/controllers/auth_controller.ex

```diff
@@ -0,0 +1,124 @@
+defmodule OrbitWeb.AuthController do
+  use OrbitWeb, :controller
+  plug(Ueberauth)
+
+  alias OrbitWeb.Auth.Auth
+  require Logger
+
+  def login_page(conn, _params) do
+    if Auth.logged_in_user(conn) do
+      redirect_login_successful(conn)
+    else
+      redirect(conn, to: ~p"/auth/keycloak")
+    end
+  end
+
+  def request(conn, _params) do
+    # unrecognized provider
+    # (recognized providers are caught by Ueberauth before this point)
+    conn
+    |> put_status(:bad_request)
+    |> text("Bad Request (unrecognized provider)")
+  end
+
+  defp get_logout_url(conn, auth) do
+    case UeberauthOidcc.initiate_logout_url(auth, %{
+           # We have to generate exactly what Keycloak/EntraID expect for logout redirect
+           post_logout_redirect_uri:
+             conn
+             |> url(~p"/login")
+             |> URI.parse()
+             # SSL is terminated before Phoenix sees requests, so we have to pretend it's https
+             # even if Phoenix thinks it isn't.
+             |> Map.put(:scheme, "https")
+             # We're on 80, but :80 appended to the URL also makes EntraID unhappy.
+             |> Map.put(:port, nil)
+             |> URI.to_string()
+         }) do
+      {:ok, logout_url} -> logout_url
+      _ -> nil
+    end
+  end
+
+  def callback(%{assigns: %{ueberauth_auth: %{provider: :keycloak} = auth}} = conn, _params) do
+    username = String.replace(auth.info.email, "MBTA.com", "mbta.com")
+    # credentials = auth.credentials
```

Comment:
**nitpick:** remaining commented-out code.

Reply:
I was hoping to leave this in whenever we get to keycloak groups/refresh tokens. But I can take it out!